packages/driver/src/cypress/mocha.ts

```diff
@@ -95,40 +146,68 @@ export function calculateTestStatus (test: CypressTest, strategy: Strategy, opti
       ((options as Options<'detect-flake-but-always-fail'>).stopIfAnyPassed || false) :
       null
 
-    // Do we have the required amount of passes? If yes, we no longer need to keep running the test.
-    if (strategy !== 'detect-flake-but-always-fail' && passingAttempts >= (passesRequired as number)) {
-      outerTestStatus = 'passed'
-      test.final = true
-      shouldAttemptsContinue = false
-    } else if (totalAttemptsAlreadyExecuted < maxAttempts &&
-      (
-        // For strategy "detect-flake-and-pass-on-threshold" or no strategy (current GA retries):
-        //  If we haven't met our max attempt limit AND we have enough remaining attempts that can satisfy the passing requirement.
-        // retry the test.
-        (strategy !== 'detect-flake-but-always-fail' && remainingAttempts >= (neededPassingAttemptsLeft as number)) ||
-        // For strategy "detect-flake-but-always-fail":
-        //  If we haven't met our max attempt limit AND
-        //    stopIfAnyPassed is false OR
-        //    stopIfAnyPassed is true and no tests have passed yet.
-        // retry the test.
-        (strategy === 'detect-flake-but-always-fail' && (!stopIfAnyPassed || stopIfAnyPassed && passingAttempts === 0))
-      )) {
-      test.final = false
+    switch (strategy) {
+      case 'detect-flake-and-pass-on-threshold':
+        if (passingAttempts >= (passesRequired as number)) {
+          // we met the threshold, so we can stop retrying and pass the test
+          outerTestStatus = 'passed'
+          test.final = true
+          shouldAttemptsContinue = false
+          test.reasonToStop = 'PASSED_MET_THRESHOLD'
```

Comment:
`reasonToStop` is only set, and never referenced - is this a field we'll be sending to the cloud?

Reply:
Yes, see the description [here](https://cypress-io.atlassian.net/browse/CYCLOUD-1665)